workerGroup/workerGroup_test.go

```diff
@@ -0,0 +1,40 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (C) 2016 VMware, Inc. All rights reserved.
+// -- VMware Confidential
+///////////////////////////////////////////////////////////////////////
+package workerGroup
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWorkerGroup(t *testing.T) {
```

Comment:
please add another test that creates 50 go routines or something and exercise the group for a while